umbraco/Umbraco-CMS
# Adds icons from App_Plugins/**/Icons/*.svg

### Prerequisites


### Description
As mentioned in https://github.com/umbraco/Umbraco-CMS/pull/5606#issuecomment-500165185 it would be good to have a way of adding icons without having to put them in /umbraco/assets/icons.

So I added the functionality to the IconController, so it can search through `~/App_Plugins/**/Icons/*.svg` for more icons to add.

This way it will be a lot easier for packages to add its own icons.

### To test
Add some icons in a folder called icons inside an App_Plugin folder.
You can download [this one](https://github.com/umbraco/Umbraco-CMS/files/5212751/CustomIcons.zip) as an example. It contains icons called `icon-allen-keys` and `lego-blocks`.

Open something using icons, ie. the doc type editor, and verify that the icons are available:
![image](https://user-images.githubusercontent.com/3726467/92996904-75ecf700-f50f-11ea-8979-b76ece6b1a43.png)
![image](https://user-images.githubusercontent.com/3726467/92996923-89985d80-f50f-11ea-939f-809a88d95df5.png)

Also verify after selecting the icon, saving the doctype, and reloading the browser window, that eg. `/umbraco/backoffice/UmbracoApi/Icon/GetIcon?iconName=icon-allen-keys` works and returns the desired icon.
